codec/address_test.go

```diff
@@ -39,3 +39,79 @@ func TestAddressJSON(t *testing.T) {
 	require.NoError(json.Unmarshal(addrJSONBytes, &parsedAddr))
 	require.Equal(addr, parsedAddr)
 }
+
+func TestEncodeWithChecksum(t *testing.T) {
+	require := require.New(t)
+
+	tests := []struct {
+		address     Address
```

Comment:
Could we add a name to each test case and rename `checksumStr` to `expectedStr` ?

Reply:
Done